runtime/auth/aws-signing-common/common/src/aws/smithy/kotlin/runtime/auth/awssigning/internal/AwsChunkedUtil.kt

```diff
@@ -28,3 +34,37 @@ internal fun SdkBuffer.writeTrailers(
         }
     writeUtf8("x-amz-trailer-signature:${signature}\r\n")
 }
+
+/**
+ * @return a boolean representing if this HttpBody is eligible to send via aws-chunked content encoding
+ */
+internal val HttpBody.isEligibleForAwsChunkedStreaming: Boolean
+    get() = (this is HttpBody.SourceContent || this is HttpBody.ChannelContent) && contentLength != null &&
+        (isOneShot || contentLength!! > AwsSigningMiddleware.AWS_CHUNKED_THRESHOLD)
+
+/**
+ * @return a boolean representing if the signing configuration is configured (via [HashSpecification]) for aws-chunked content encoding
+ */
+internal val AwsSigningConfig.useAwsChunkedEncoding: Boolean
+    get() = when (hashSpecification) {
+        is HashSpecification.StreamingAws4HmacSha256Payload, is HashSpecification.StreamingAws4HmacSha256PayloadWithTrailers -> true
+        else -> false
+    }
+
+/**
+ * Set the HTTP headers required for the aws-chunked content encoding
+ */
+internal fun HttpRequestBuilder.setAwsChunkedHeaders() {
+    headers.setMissing("Content-Encoding", "aws-chunked")
+    headers.setMissing("Transfer-Encoding", "chunked")
+    headers.setMissing("X-Amz-Decoded-Content-Length", body.contentLength!!.toString())
+}
+
+internal expect fun HttpRequestBuilder.internalSetAwsChunkedBody(signer: AwsSigner, signingConfig: AwsSigningConfig, signature: ByteArray)
+
+/**
+ * Update the HTTP body to use aws-chunked content encoding
+ */
+internal fun HttpRequestBuilder.setAwsChunkedBody(signer: AwsSigner, signingConfig: AwsSigningConfig, signature: ByteArray) {
```

Comment:
question: why the redirect to `internalSetAwsChunkedBody`? why not just make this function `expect/actual` instead?

Reply:
discussed offline, fixed the NoSuchMethodError with a clean build >:)